src/hooks/useObject.ts

```diff
@@ -0,0 +1,128 @@
+import { FormContextValues } from 'react-hook-form'
+
+import {
+  UseObjectProperties,
+  UseObjectReturnType,
+  BasicInputReturnType,
+  UISchemaType,
+  UITypes,
+} from './types'
+import {
+  JSONSchemaType,
+  useObjectFromPath,
+  concatFormPath,
+} from '../JSONSchema'
+import { getGenericInput } from './useGenericInput'
+import { getInputCustomFields } from './useInput'
+import { getRadioCustomFields } from './useRadio'
+import { useFormContext } from '../components'
+import { getSelectCustomFields } from './useSelect'
+import { getHiddenCustomFields } from './useHidden'
+import { getPasswordCustomFields } from './usePassword'
+import { getTextAreaCustomFields } from './useTextArea'
+
+function getFromGeneric(
+  genericInput: BasicInputReturnType
+): UseObjectReturnType {
+  const currentObject = genericInput.getObject()
+  const inputs: UseObjectReturnType = []
+
+  switch (currentObject.type) {
+    case 'string':
+      if (currentObject.enum) {
+        inputs.push(getSelectCustomFields(genericInput))
+      } else {
+        inputs.push(getInputCustomFields(genericInput))
+      }
+      break
+    case 'integer':
+    case 'number':
+      inputs.push(getInputCustomFields(genericInput))
+      break
+    case 'boolean':
+      inputs.push(getRadioCustomFields(genericInput))
+      break
+  }
+  return inputs
+}
+
+function getStructure(
+  formContext: FormContextValues,
+  pathInfo: [JSONSchemaType, boolean, string],
+  path: string,
+  UISchema: UISchemaType | undefined
+): UseObjectReturnType {
+  let inputs: UseObjectReturnType = []
+  const currentObject = pathInfo[0]
+
+  const genericInput = getGenericInput(formContext, pathInfo, path)
+
+  if (currentObject.type === 'object') {
+    const objKeys = Object.keys(currentObject.properties)
+    const requiredFields: Array<string> | undefined = currentObject.required
+    for (const key of objKeys) {
+      const isRequired = requiredFields
+        ? requiredFields.indexOf(key) !== -1
+        : false
+
+      const currentPathInfo: [JSONSchemaType, boolean, string] = [
```

Comment:
I saw this type (`[JSONSchemaTYpe, boolean, string]` repeated some times, maybe it would be better if you create a type for it instead of repeating it